packages/material-ui/src/SpeedDial/SpeedDial.test.js

```diff
@@ -225,48 +228,47 @@ describe('<SpeedDial />', () => {
     };
 
     const resetDialToOpen = (direction) => {
-      if (wrapper && wrapper.exists()) {
-        wrapper.unmount();
-      }
+      unmount();
 
-      mountSpeedDial(direction);
+      renderSpeedDial(direction);
       dialButtonRef.focus();
     };
 
     it('displays the actions on focus gain', () => {
       resetDialToOpen();
-      expect(wrapper.find(SpeedDial).props().open).to.equal(true);
+      expect(screen.getAllByRole('menuitem')).to.have.lengthOf(4);
+      expect(screen.getByRole('menu')).not.to.have.class(classes['actionsClosed']);
     });
 
     describe('first item selection', () => {
       it('considers arrow keys with the same initial orientation', () => {
         resetDialToOpen();
-        getDialButton().simulate('keydown', { key: 'left' });
+        fireEvent.keyDown(getDialButton(), { key: 'left' });
         expect(isActionFocused(0)).to.equal(true);
-        getDialButton().simulate('keydown', { key: 'up' });
+        fireEvent.keyDown(getActionButton(0), { key: 'up' });
         expect(isActionFocused(0)).to.equal(true);
-        getDialButton().simulate('keydown', { key: 'left' });
+        fireEvent.keyDown(getActionButton(0), { key: 'left' });
         expect(isActionFocused(1)).to.equal(true);
-        getDialButton().simulate('keydown', { key: 'right' });
+        fireEvent.keyDown(getActionButton(1), { key: 'right' });
         expect(isActionFocused(0)).to.equal(true);
       });
     });
 
     // eslint-disable-next-line func-names
     describe('actions navigation', function () {
-      this.timeout(5000); // These tests are really slow.
+      this.timeout(10000); // These tests are really slow.
```

Comment:
Is this needed? It's scary if it needs 10s to run.

Reply:
It timed out in a previous commit on CircleCI, I can reset it back and see if it passes 😄

Reply:
<img width="1061" alt="Screenshot 2020-10-14 at 14 07 05" src="https://user-images.githubusercontent.com/12977174/95993160-abdf0d00-0e26-11eb-97c1-0185c812e9a6.png">
Reset back to 5s and it fails, I will increase it to 7.5s

Reply:
Ok, in this case, I think that we should rethink the test. I don't think that it's for one test to take 7s.

Reply:
I think its because before each testCombination call, the tests reset the speed dial by cleaning up and rendering it again (or the old way of unmounting and mounting). The best way would be just to split up the testCombination into individual `it` calls

Reply:
@oliviertassinari I have updated the tests to split them into individual tests and remove the timeout call. Hope this is ok now 😃